pontoon/sync/changeset.py

```diff
@@ -190,70 +190,84 @@ def execute_create_db(self):
+
+            # Modify existing translation.
             if db_translation:
                 if not db_translation.approved and not vcs_translation.fuzzy:
                     db_translation.approved = True
-                    db_translation.approved_date = self.now
                     db_translation.approved_user = user
+                    db_translation.approved_date = self.now
                 db_translation.rejected = False
                 db_translation.fuzzy = vcs_translation.fuzzy
                 db_translation.extra = vcs_translation.extra
 
                 if db_translation.is_dirty():
                     self.translations_to_update[db_translation.pk] = db_translation
-                if not db_translation.fuzzy:
-                    approved_translations.append(db_translation)
-                else:
+                if db_translation.fuzzy:
                     fuzzy_translations.append(db_translation)
+                else:
+                    approved_translations.append(db_translation)
+
+            # Create new translation.
             else:
                 self.translations_to_create.append(Translation(
                     entity=db_entity,
                     locale=self.locales[locale_code],
                     string=string,
                     plural_form=plural_form,
                     approved=not vcs_translation.fuzzy,
-                    approved_date=self.now if not vcs_translation.fuzzy else None,
                     approved_user=user,
+                    approved_date=self.now if not vcs_translation.fuzzy else None,
                     user=user,
                     fuzzy=vcs_translation.fuzzy,
                     extra=vcs_translation.extra
                 ))
 
-        # Any existing translations that were not approved get unapproved.
-        for translation in old_translations:
+        # Unapprove translations that were approved before the sync job started unless sync
+        # resolves them as active approved translations.
+        # Note: If translations get approved after the sync starts, duplicate approvals can arise.
+        # We take care of that at the and of the sync job in tasks.py.
+        for translation in db_translations_approved_before_sync:
             if translation not in approved_translations:
-                # Use the translation instance already set for update if it exists
+                # Use the translation instance already set for update if it exists.
                 translation = self.translations_to_update.get(translation.pk, translation)
                 translation.approved = False
                 translation.approved_user = None
                 translation.approved_date = None
 
+                # Reject translations unless they became fuzzy during sync. Condition is sufficient
+                # because they were approved previously.
+                if not translation.fuzzy:
+                    translation.rejected = True
```

Comment:
nit: maybe we could create a general methods for such operations? e.g. translatios.reject(user), translation.approve?